app/components/elements/image/index.js

```diff
@@ -0,0 +1,20 @@
+import {pick, find, map} from 'wasmuth'
+import {dispatch, set} from '/store'
+import mapStateToProps from '/util/mapStateToProps'
+import Base from './base'
+
+export const Image = mapStateToProps(
+  ({loadedImages = {}}, {images}) => ({
+    loadedImages: pick(map(i => i.url, images), loadedImages)
+  })
+)(({images, loadedImages}) => {
+  const bestAvailableImage = find(i => !!loadedImages[i.url], images.slice().reverse()) || {}
```

Comment:
Should highest res be the first in the array? Then avoid reverse? Oh, I guess the order in the DOM needs to be the reverse anyway?

What about a quality key (Number) on the image object? Seems like it could be easy to forgot the order the array should take. But, a quick comment on the code could be enough?